js/runner.go

```diff
@@ -504,6 +504,9 @@ func (u *VU) runFn(
 	if opts.SystemTags.Has(stats.TagGroup) {
 		tags["group"] = group.Path
 	}
+	if scenario != "" && opts.SystemTags.Has(stats.TagScenario) {
+		tags["scenario"] = scenario
+	}
```

Comment:
So we don't want to let people overwrite `scenario` as a tag ?

Reply:
hmm :confused: what's stopping them from overwriting it in the per-request `tags`?

Reply:
we merge them above in `tags` and here we set `tags[scenario]` regardless :). Same btw goes for iter, vu and group. To be honest, I am not against us overwriting them, but maybe at least have some log about it. 

Although this will probably need to be as part of #1478 as here it will be too noisy

Reply:
hmm the tags we merge with above are the VU activation tags, I think... so yes, I don't think you should be able to overwrite the k6 system tags with the per-scenario tags

Reply:
Yeah, I also considered it a deliberate design choice, especially since there's a workaround: they can create a custom tag with the same name and use `--system-tags` to disable the built-in ones. So I'm OK with this approach too.

Reply:
Yeah, I am also okay with us overwriting it. I would just prefer if we log it somewhere when it happens. 

With #1478 I can do it once per initialization ... which is probably way too noisy still ... :(

Reply:
but you can (and should) just merge this for now :D